src/terminal.rs

```diff
@@ -12,13 +12,40 @@ pub struct Terminal {
 
 impl Terminal {
     pub fn new() -> Self {
-        Terminal {
+        Self {
             input: Term::stdout(),
             output: Term::stdout(),
         }
     }
+}
+
+pub trait UserInterface {
+    fn get_user_command(&self) -> Result<UserCommand, TerminalError>;
+    fn show_options(&self) -> Result<(), TerminalError>;
+    fn finish_todo(&self) -> Result<(), TerminalError>;
+    fn show_invalid_option(&self) -> Result<(), TerminalError>;
+    fn ask_for_new_todo(&self) -> Result<Todo, TerminalError>;
+    fn show_todo(&self, todo: &Todo, msg_initial: &str) -> Result<(), TerminalError>;
+    fn show_error(&self, error: TerminalError);
+    fn get_todo_for_update(
+        &self,
+        list_todos: &Box<dyn TodoStorage>,
```

Comment:
```suggestion
        list_todos: &dyn TodoStorage,
```

Na maioria das situações, `&Box<_>` é desnecessário.

Reply:
Certo! Para remover o `&Box<_>` eu precisei desreferenciar na chamada ao passar por parâmetro  `check_list_is_empty(&*self.todo_storage)`. Isso está correto?

Reply:
Acredito que apenas `check_list_is_empty(&self.todo_storage)` já seria suficiente, mas esse tipo de coisa o linter vai indicar quando chegarmos nessa parte 😉

Reply:
Beleza!!